services/task-svc/internal/bed/bed.go

```diff
@@ -0,0 +1,220 @@
+		RoomId: bed.RoomID.String(),
+	}, nil
+}
+
+func (ServiceServer) GetBeds(ctx context.Context, _ *pb.GetBedsRequest) (*pb.GetBedsResponse, error) {
+	db := hwgorm.GetDB(ctx)
+
+	organizationID, err := common.GetOrganizationID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var beds []Bed
+	if err := db.Where("organization_id = ?", organizationID.String()).Find(&beds).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, "id not found")
+		}
+	}
+
+	res := pb.GetBedsResponse{
+		Beds: []*pb.GetBedsResponse_Bed{},
+	}
+
+	for _, bed := range beds {
+		res.Beds = append(res.Beds, &pb.GetBedsResponse_Bed{
+			Id:     bed.ID.String(),
+			RoomId: bed.RoomID.String(),
+		})
+	}
+
+	return &res, nil
+}
+
+func (ServiceServer) GetBedsByRoom(ctx context.Context, req *pb.GetBedsByRoomRequest) (*pb.GetBedsByRoomResponse, error) {
+	db := hwgorm.GetDB(ctx)
+
+	organizationID, err := common.GetOrganizationID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	roomID, err := uuid.Parse(req.RoomId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	var beds []Bed
+	if err := db.Where("organization_id = ? AND room_id = ?", organizationID.String(), roomID.String()).Find(&beds).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, "roomID not found")
+		}
+	}
+
+	res := pb.GetBedsByRoomResponse{
+		Beds: []*pb.GetBedsByRoomResponse_Bed{},
+	}
+
+	for _, bed := range beds {
+		res.Beds = append(res.Beds, &pb.GetBedsByRoomResponse_Bed{
+			Id: bed.ID.String(),
+		})
+	}
+
+	return &res, nil
+}
+
+func (ServiceServer) UpdateBed(ctx context.Context, req *pb.UpdateBedRequest) (*pb.UpdateBedResponse, error) {
+	db := hwgorm.GetDB(ctx)
+
+	bedID, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	bed := models.Bed{ID: bedID}
+	updates := req.UpdatesMap()
+	// TODO: respect req.RoomID in this or another method
+
+	if err := db.Model(&bed).Updates(updates).Error; err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	return &pb.UpdateBedResponse{}, nil
+}
+
+func (ServiceServer) DeleteBed(ctx context.Context, req *pb.DeleteBedRequest) (*pb.DeleteBedResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	organizationID, err := common.GetOrganizationID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	bedID, err := uuid.Parse(req.Id)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	bed := Bed{ID: bedID}
+	if err := db.Where("organization_id = ?", organizationID).First(&bed).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, "id not found")
```

Comment:
Matter of opinion, but the HTTP standard agrees with me here: DELETEs should be idempotent

Reply:
We aren't using the http protocol directly. If so we shouldn't send any errors for GET and PUT, too.